app/code/community/Bolt/Boltpay/Model/Cron.php

```diff
@@ -44,4 +44,44 @@ public function cleanupQuotes() {
 
         $connection->query($sql);
     }
+
+    /**
+     * After a pre-auth pending order has existed for 15 minutes or more, we remove it from the system.
+     * At this point, Bolt should have called the "failed_payment" webhook to do this. We are catching
+     * the anomaly cases where this mechanism fails.
+     */
+    public function cleanupOrders() {
+        $expiration_time = Mage::getSingleton('core/date')->gmtDate('Y-m-d H:i:s', time()-(60*15));  // Magento uses GMT to save in DB
+
+        /* @var Mage_Sales_Model_Resource_Order_Collection $orderCollection */
+        $orderCollection = Mage::getModel('sales/order')->getCollection();
+
+        /** @var Mage_Sales_Model_Order $deletePendingPaymentOrdersBeforeThis */
+        $deletePendingPaymentOrdersBeforeThis = $orderCollection
+            ->addFieldToFilter('created_at', array( 'gte' => $expiration_time))
+            ->setOrder('created_at', 'ASC')
+            ->getFirstItem();
+
+        /* @var Mage_Sales_Model_Resource_Order_Collection $expiredPendindOrderCollection */
+        $expiredPendingPaymentOrderCollection = Mage::getModel('sales/order')->getCollection();
+        $expiredPendingPaymentOrderCollection
+            ->addFieldToFilter('entity_id', array( 'lt' => $deletePendingPaymentOrdersBeforeThis->getId()))
+            ->addFieldToFilter('status', Bolt_Boltpay_Model_Payment::TRANSACTION_PRE_AUTH_PENDING)
+        ;
+
+        $ordersToRemove = $expiredPendingPaymentOrderCollection->getItems();
+
+        /** @var Bolt_Boltpay_Model_Order $orderModel */
+        $orderModel = Mage::getModel('boltpay/order');
+
+        /** @var Mage_Sales_Model_Order $order */
+        foreach($ordersToRemove as $order) {
+            try {
+                $orderModel->removePreAuthOrder($order);
```

Comment:
do we need separate treatment for this case? in failed_payment case we want to reactivate quote but in 500 case it's not (?)

Reply:
@daisy1754 I don't understand your reasoning here.  I don't see a separate case.  There is only one case here: *if an order is still in the pending_bolt state after an unreasonable amount of time.*  In all cases, these orders should be handled regardless of under which conditions the order was created ( 500, timeout, abnormal response.)  How do you see this differently?

Reply:
tbh I didn't too much thought (it's more of question than suggestion). I think what you say makes sense